resources/codegen/python/openassetio_codegen/generators/helpers.py

```diff
@@ -0,0 +1,77 @@
+#
+#   Copyright 2022 The Foundry Visionmongers Ltd
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+"""
+Utility functions to help code generation templates.
+"""
+import datetime
+import re
+
+from typing import List, Union
+from ..datamodel import SpecificationDeclaration, TraitDeclaration
+
+
+def to_upper_camel_alnum(string: str) -> str:
+    """
+    Reformats a string to UpperCamelCase stripping out any non
+    alpha-numeric characters.
+
+    eg
+    : "ThisIs! a complex-string" -> "ThisIsAComplexString"
+    """
+    words = re.split("([^a-zA-Z0-9]|[A-Z](?:[a-z]+)|[A-Z](?=[A-z]|$))", string)
+    legal_words = [word.capitalize() for word in words if (word and word.isalnum())]
```

Comment:
:confused:  Try as I might, I can't break it, so happy to leave well alone.

Reply:
Be good to verify [this](https://github.com/OpenAssetIO/OpenAssetIO/pull/529/commits/4cb91a7a43a100057bd9b8c5092cd9ff37586dad)